scoring/score.py

```diff
@@ -1,20 +1,64 @@
+# -*- coding: utf-8 -*-
+
+import itertools
 import collections
 
 
 class InvalidScoresheetException(Exception):
     pass
 
 
+POINTS_HELD = 1
+POINTS_GROUND = 2
+POINTS_PLATFORM = 5
+
+NUM_TOKENS = 16
+
+
 class Scorer(object):
     def __init__(self, teams_data, arena_data):
         self._teams_data = teams_data
         self._arena_data = arena_data
 
     def validate(self, extra):
-        pass
+        token_counts = tuple(itertools.chain(
+            [x.get('tokens_held', 0) for x in self._teams_data.values()],
+            [extra.get('tokens_ground', 0)],
+            [x.get('tokens_ground', 0) for x in self._arena_data.values()],
+            [x.get('tokens_platform', 0) for x in self._arena_data.values()],
+        ))
```

Comment:
`tuple` on `itertools.chain` here is probably less clear than just using `+` on the four lists.

Reply:
2ab7aab